templates/pkg/resource/resource.go.tpl

```diff
@@ -80,6 +80,9 @@ func (r *resource) SetStatus(desired acktypes.AWSResource) {
 // resource identifier
 func (r *resource) SetIdentifiers(identifier *ackv1alpha1.AWSIdentifiers) error {
 {{- GoCodeSetResourceIdentifiers .CRD "identifier" "r.ko" 1}}
+{{- if $hookCode := Hook .CRD "set_identifiers_pre_read_one" }}
```

Comment:
The position of hook is not pre read one. There is no read one call here.

I agree this can be a useful hook. maybe rename it to just "set_identifiers"

Just a thought, Should there be either hook code OR auto-generated code inside method body? 
This is such a simple method, no need to have both.